lib/beaker-puppet/helpers/setup/aio.rb

```diff
@@ -0,0 +1,112 @@
+module Beaker
+  module DSL
+    module Helpers
+      module Setup
+        module AIO
+          require 'beaker-puppet/install_utils/puppet5'
+          require 'beaker-puppet/install_utils/foss_utils'
+          require 'beaker-puppet/helpers/setup/common'
+
+          include BeakerPuppet::Install::Puppet5
+          include Beaker::DSL::InstallUtils::FOSSUtils
+          include Beaker::DSL::Helpers::Setup::Common
+
+          def aio_install
+            test_name "Installl Packages" do
+              dev_builds_url  = ENV['DEV_BUILDS_URL'] || 'http://builds.delivery.puppetlabs.net'
+
+              step "Install puppet-agent..." do
+                sha = ENV['SHA']
+                install_from_build_data_url('puppet-agent', "#{dev_builds_url}/puppet-agent/#{sha}/artifacts/#{sha}.yaml", hosts)
+              end
+
+              step "Install puppetserver..." do
+                if ENV['SERVER_VERSION'].nil? || ENV['SERVER_VERSION'] == 'latest'
+                  install_puppetlabs_dev_repo(master, 'puppetserver', 'latest', nil, :dev_builds_url => "http://nightlies.puppet.com")
+                  master.install_package('puppetserver')
+                else
+                  server_version = ENV['SERVER_VERSION']
+                  install_from_build_data_url('puppetserver', "#{dev_builds_url}/puppetserver/#{server_version}/artifacts/#{server_version}.yaml", master)
+                end
+              end
+
+              # make sure install is sane, beaker has already added puppet and ruby
+              # to PATH in ~/.ssh/environment
+              hosts.each do |host|
+                on host, puppet('--version')
+                ruby = ruby_command(host)
+                on host, "#{ruby} --version"
+              end
+
+              # Get a rough estimate of clock skew among hosts
+              times = []
+              hosts.each do |host|
+                ruby = ruby_command(host)
+                on(host, "#{ruby} -e 'puts Time.now.strftime(\"%Y-%m-%d %T.%L %z\")'") do |result|
+                  times << result.stdout.chomp
+                end
+              end
+              times.map! do |time|
+                (Time.strptime(time, "%Y-%m-%d %T.%L %z").to_f * 1000.0).to_i
+              end
+              diff = times.max - times.min
+              if diff < 60000
+                logger.info "Host times vary #{diff} ms"
+              else
+                logger.warn "Host times vary #{diff} ms, tests may fail"
+              end
+
+              configure_gem_mirror(hosts)
+            end
+          end
+
+          def install_cumulus_modules
+            test_name "Install Cumulus Modules" do
+              platforms = hosts.map{|val| val[:platform]}
+              skip_test "No cumulus hosts present" unless platforms.any? { |val| /cumulus/ =~ val }
```

Comment:
@joshcooper it looks like this skip is causing the top level test (https://github.com/puppetlabs/puppet/pull/6602/files#diff-62e97df027f65c15385cb6a34498e2acR1) to skip, unless I'm missing something? Is this something that happens with nested tests?

```
  Install Cumulus Modules

    No cumulus hosts present
Begin teardown
End teardown
Warning: setup/aio.rb skipped in 91.94 seconds
      Test Suite: pre_suite @ 2018-02-06 11:10:42 -0800

      - Host Configuration Summary -


              - Test Case Summary for suite 'pre_suite' -
       Total Suite Time: 91.94 seconds
      Average Test Time: 91.94 seconds
              Attempted: 1
                 Passed: 0
                 Failed: 0
                Errored: 0
                Skipped: 1
                Pending: 0
                  Total: 1

      - Specific Test Case Status -

Failed Tests Cases:
Errored Tests Cases:
Skipped Tests Cases:
  Test Case setup/aio.rb
Pending Tests Cases:

No tests to run for suite 'tests'
No tests to run for suite 'post_suite'
No tests to run for suite 'pre_cleanup'
Cleanup: cleaning up after successful run
```

Reply:
alright, I got this figured out